### PR TITLE
Add modulemd-stream v3

### PR DESCRIPTION
**Pull request by mmathesius**
Initial implementation of new ModuleStreamV3 object, parsing/emitting of `modulemd-stream` v3 documents, and a basic parse/dump test.

### CODE REVIEW

**Comment by mmathesius** [comment]
CC: @mcurlej

File: modulemd/modulemd-module-stream.c [hunk 1]
```diff
@@ -472,7 +494,8 @@ modulemd_module_stream_upgrade (ModulemdModuleStream *self,
   if (!mdversion)
     {
       /* If target mdversion is unspecified, set it to the latest */
-      mdversion = MD_MODULESTREAM_VERSION_LATEST;
+      /* TODO: reset to MD_MODULESTREAM_VERSION_LATEST */
+      mdversion = MD_MODULESTREAM_VERSION_TWO;
     }
```
Comment (sgallagher) [inline modulemd/modulemd-module-stream.c:499]: Should this TODO be updated?

File: modulemd/modulemd-module-stream.c [hunk 2]
```diff
@@ -1179,7 +1202,8 @@ modulemd_module_stream_class_init (ModulemdModuleStreamClass *klass)
     "Metadata Version",
     "The metadata version of this ModuleStream object. Read-only.",
     0,
-    MD_MODULESTREAM_VERSION_LATEST,
+    /* TODO: reset to MD_MODULESTREAM_VERSION_LATEST */
+    MD_MODULESTREAM_VERSION_TWO,
     0,
     G_PARAM_READABLE | G_PARAM_STATIC_STRINGS);
```
Comment (sgallagher) [inline modulemd/modulemd-module-stream.c:1208]: Same

File: modulemd/tests/ModulemdTests/modulestream.py [hunk 3]
```diff
@@ -643,6 +647,14 @@ def test_upgrade(self):
...
         )
 
+    def test_upgrade_v2_to_v3(self):
+        # TODO: implement test
+        pass
+
+    def test_upgrade_v1_to_v3(self):
+        # TODO: implement test
+        pass
+
```
Comment (sgallagher) [inline modulemd/tests/ModulemdTests/modulestream.py:657]: If the upgrade isn't implemented yet, you should probably leave the test out of the PR.

**Review by sgallagher — CHANGES_REQUESTED** [review]

File: modulemd/modulemd-module-stream.c (hunk 1 above)
Comment (mmathesius) [reply modulemd/modulemd-module-stream.c]: Yes, thanks for catching the temporary tweak that can now be reverted.

File: modulemd/modulemd-module-stream.c (hunk 2 above)
Comment (mmathesius) [reply modulemd/modulemd-module-stream.c]: Reverted.

File: modulemd/tests/ModulemdTests/modulestream.py (hunk 3 above)
Comment (mmathesius) [reply modulemd/tests/ModulemdTests/modulestream.py]: These were placeholders to make sure I didn't forget about them. I'll include them with my upgrade work instead. I also dropped the corresponding C versions of the placeholders from this PR.

**Review by sgallagher — APPROVED** [review]
LGTM

**Comment by mmathesius** [comment]
I just added one small improvement for context validation. Assuming you're good with it, can we go ahead and squash these commits and merge into the devel branch?

**Comment by sgallagher** [comment]
> I just added one small improvement for context validation. Assuming you're good with it, can we go ahead and squash these commits and merge into the devel branch?

Go ahead.

**Comment by mmathesius** [comment]
> > I just added one small improvement for context validation. Assuming you're good with it, can we go ahead and squash these commits and merge into the devel branch?
> 
> Go ahead.

Thanks, @sgallagher. Squashed and rebased with the OpenMandriva workaround. I'll merge when I confirm CI passes.

**Pull request merged**